### PR TITLE
Enable external settings of properties

### PR DESCRIPTION
**Pull request by ghost**
As requested on the forum, allows properties of menu items (icons, thumbs, backgrounds, widgets, any custom properties) to be set from outside of the management dialog.

@marcelveldt if you have time to cast an eye over it would be appreciated. Also, are you happy with what the PR does - this continues the script down the road started by the context addon changes, whereby more can be edited from outside of the menu, and I suspect requests to change more from outside will follow.

### CODE REVIEW

**Comment by ghost** [comment]
Had confirmation from the forums there is still a bug after menu's are reset.

**Comment by ghost** [comment]
Issue found on menu reset. Code should now be working.

File: resources/lib/nodefunctions.py [hunk 1]
```diff
@@ -435,6 +435,115 @@ def extractID( self, path ):
...
+            # If the labelID isn't in the allProps[ group ], add it
+            if currentProperty[ 1 ] not in allProps[ currentProperty[ 0 ] ].keys():
+                allProps[ currentProperty[ 0 ] ][ currentProperty[ 1 ] ] = {}
+            # And add the property to allProps[ group ][ labelID ]
+            if currentProperty[ 3 ] is not None:
+                allProps[ currentProperty[ 0 ] ][ currentProperty [ 1 ] ][ currentProperty[ 2 ] ] = currentProperty[ 3 ]
+
+        # Loop through the properties we've been asked to set
+        count = -1
+        for propertyName in propertyNames:
```
Comment (phil65) [inline resources/lib/nodefunctions.py]: for count, propertyName in enumerate(propertyNames)
would be the pythonic way for any kind of loop with a counter. That way you dont have to initalize and increment manually.

**Comment by ghost** [comment]
Cheers Phil. I learn something new every day :)

**Pull request merged**